src/common/components/layouts/pages/inspect/tree/Tree.js

```diff
@@ -60,20 +60,29 @@ export const errorMessagesMap = {
     },
 };
 
-function Tree({ ruleSummaries, selectedCheck, setSelectedCheck, errorsMap, profile }) {
+function Tree({
+    ruleSummaries,
+    expandedRule,
+    setExpandedRule,
+    selectedCheck,
+    setSelectedCheck,
+    selectedCheckId,
+    setSelectedCheckId,
+    errorsMap,
+    profile,
+}) {
     const [language, setLanguage] = useState(getItem(LS_ERROR_MESSAGES_LANGUAGE) || languageEnum.English);
     const [anchorMenuEl, setAnchorMenuEl] = useState(null);
-    const [expandedRule, setExpandedRule] = useState(UNSELECTED);
     const prevSelectedCheck = usePrevious(selectedCheck);
     const onRuleClick = useCallback(
         index => {
-            if (expandedRule === index) {
-                return setExpandedRule(UNSELECTED);
-            }
-
-            return setExpandedRule(index);
+            const copyExpandedRule = JSON.parse(JSON.stringify(expandedRule));
```

Comment:
Reuse the method passed in props instead of code duplication (See my very first comment)